delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/wiring/ProjectsModule.scala

```diff
@@ -44,20 +68,78 @@ object ProjectsModule extends ModuleDef {
         quotas: Quotas,
         baseUri: BaseUri,
         as: ActorSystem[Nothing],
-        clock: Clock[UIO],
         uuidF: UUIDF,
         scheduler: Scheduler,
         scopeInitializations: Set[ScopeInitialization],
-        mappings: ApiMappingsCollection
+        mappings: ApiMappingsCollection,
+        cache: ProjectsCache,
+        deletionCache: DeletionStatusCache,
+        agg: ProjectsAggregate
     ) =>
       ProjectsImpl(
+        agg,
         config.projects,
         eventLog,
         organizations,
         quotas,
         scopeInitializations,
-        mappings.merge
-      )(baseUri, uuidF, as, scheduler, clock)
+        mappings.merge,
+        cache,
+        deletionCache
+      )(baseUri, uuidF, as, scheduler)
+  }
+
+  many[ResourcesDeletionWithPriority].add {
+    (cache: ProjectsCache, agg: ProjectsAggregate, dbCleanup: DatabaseCleanup) =>
+      // This priority has to be bigger than any other module, since we want projects to be handled after other resources
+      100 -> ProjectDeletion(cache, agg, dbCleanup)
```

Comment:
How do you append the project one to the end, if you just collect a set of `ResourcesDeletion`? I think a priority here doesn't hurt, since we can control what comes first and what last

Reply:
The thing is Project will always come last.
And a way to do is to append the project part later and append it after you sort all the others